compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -3697,15 +3771,16 @@ private boolean inferVariableOrClassIsolation(Set<BType> publiclyExposedObjectTy
 
                 if (Symbols.isFlagOn(symbol.flags, Flags.FINAL)) {
                     return isSubTypeOfReadOnlyOrIsolatedObjectUnionWithInference(publiclyExposedObjectTypes,
-                            classDefinitions, true, symbol.type, unresolvedSymbols);
+                            classDefinitions, moduleLevelVariables, true, symbol.type, unresolvedSymbols);
                 }
 
                 return false;
             }
 
             for (BType bType : inferenceInfo.typesOfFinalFieldsAccessedOutsideLock) {
                 if (!isSubTypeOfReadOnlyOrIsolatedObjectUnionWithInference(publiclyExposedObjectTypes, classDefinitions,
-                        true, bType, unresolvedSymbols)) {
+                                                                           moduleLevelVariables, true, bType,
+                                                                           unresolvedSymbols)) {
```

Comment:
Formatting issue

Reply:
It'a aligned with the start of the arguments in the previous line.